Improved integration with GitHub and other types of projects

brazabr opened the pull request:
- Added support for HTTP Basic Authentication (can be used with Personal Access Tokens to access private GitHub repositories)
- Added option to skip POT generation and import an existing file from repository/archive
- Added option to override GitHub branch or tag
- Modified settings page for better user experience (similar to standard WordPress inline edit)
- Improved settings page self-explanation using placeholders in fields
- Fixed main plugin file to pass validations on: `WordPress-Core`, `WordPress-Extra` and `WordPress-Docs`.

brazabr committed:
Improved integration with GitHub and other types of projects
Added support for HTTP Basic Authentication (can be used with Personal Access Tokens to access private GitHub repositories)
Added option to skip POT generation and import an existing file from repository/archive
Added option to override GitHub branch or tag
Modified settings page for better user experience (more similar to standard WordPress edit pages)
Improved settings page self-explanation using placeholders in fields

brazabr committed:
Fixed notices

brazabr committed:
Replace direct access to superglobal POST with filter_input

brazabr committed:
Converting spaces to tabs in identation

brazabr committed:
Fixed makepot meta position in code after rebase

brazabr committed:
Fixed notices when running Auto Extract from Project Actions

brazabr committed:
Implemented coding standard fixes

brazabr committed:
Fixed new project settings not coming empty

toolstack commented on gp-auto-extract.php:
```diff
@@ -77,241 +153,501 @@ public function auto_extract( $project_path ) {
+							</div>
+						</fieldset>
+						<p class="submit inline-edit-save">
+							<button type="button" class="button cancel alignleft" data-project-id="<? echo esc_attr( $project->id ); ?>"><?php esc_html_e( 'Cancel' ); ?></button>
```
Can't use a short tag (<?) in the data-project-id attribute.

brazabr replied:
Typo mistake. Fixed.

toolstack commented on gp-auto-extract.php:
```diff
@@ -77,241 +153,501 @@ public function auto_extract( $project_path ) {
+									<span class="title"><?php esc_html_e( 'Source Type' ); ?></span>
+									<select class="source_type" name="source_type_<?php echo esc_attr( $project->id ); ?>" id="source_type_<?php echo esc_attr( $project->id ); ?>">
+									<?php foreach ( $this->source_types as $id => $type ) { ?>
+										<option value="<?php echo esc_attr( $id ); ?>" <?php echo selected( $source_type, $id ); ?>><?echo esc_html( $type ); ?></option>;
```
Can't use a short tag (<?) in the option output attribute.

brazabr replied:
Typo mistake. Fixed.

toolstack commented on gp-auto-extract.php:
```diff
@@ -77,241 +153,501 @@ public function auto_extract( $project_path ) {
+				esc_attr( $project->id ),
+				/* translators: %s: project name */
+				esc_attr( sprintf( __( 'Reset &#8220;%s&#8221;' ), $project->name ) ),
+				esc_html__( 'Reset' )
```
The reset option needs a confirmation alert before executing.

toolstack commented on gp-auto-extract.php:
```diff
@@ -77,241 +153,501 @@ public function auto_extract( $project_path ) {
+									<label class="alignleft">
+										<span class="title"><?php esc_html_e( 'Format' ); ?></span>
+										<?php
+										$format_options = array();
```
Should support auto-detection as the default.

brazabr replied:
Implemented in new commit.

toolstack commented on gp-auto-extract.php:
```diff
@@ -21,51 +23,125 @@
+	 * This function is here as placeholder to support adding the auto extract option to the router.
+	 * Without this placeholder there is a fatal error generated.
+	 */
+	public function before_request() {
```
This placeholder function was removed in the current tree, you must have added it back in by accident during the rebase.

brazabr replied:
Removed in recent commits.

toolstack commented on gp-auto-extract.php:
```diff
@@ -77,241 +153,501 @@ public function auto_extract( $project_path ) {
+	 * This function is here as placeholder to support adding the auto extract option to the router.
+	 * Without this placeholder there is a fatal error generated.
+	 */
+	public function after_request() {
```
This placeholder function was removed in the current tree, you must have added it back in by accident during the rebase.

brazabr replied:
Removed in recent commits.

toolstack commented on gp-auto-extract.php:
```diff
@@ -77,241 +153,501 @@ public function auto_extract( $project_path ) {
+					$project_settings[ $project->id ]['branch']              = filter_input( INPUT_POST, 'branch_' . $project->id );
+					$project_settings[ $project->id ]['use_http_basic_auth'] = filter_input( INPUT_POST, 'use_http_basic_auth_' . $project->id );
+					$project_settings[ $project->id ]['http_auth_username']  = filter_input( INPUT_POST, 'http_auth_username_' . $project->id );
+					$project_settings[ $project->id ]['http_auth_password']  = filter_input( INPUT_POST, 'http_auth_password_' . $project->id );
```
Storing the password in clear text in the database is a no-no.

toolstack reviewed (changes requested):
You've used esc_html_e() (and related escaping functions) in places you don't need them and in fact sometimes they would break the display if a translation used some html in them.

brazabr committed:
Added settings file to help coding standard

brazabr committed:
Fixed short tag errors.

Limited escaping functions to echo of user inputs.

Fixed rebase issues.

brazabr committed:
Added auto-detect support.

Fixed class for error messages.

brazabr committed:
Update url templates with fixed position parameters.

brazabr committed:
Added confirmation popup before reset.

brazabr committed:
Removed duplicated event handler.

brazabr committed:
Added encryption to saved password.

User can override default encryption key and cypher using constants in the wp-config.php if they need to.

brazabr committed:
Moved password decryption in extract method.

brazabr committed:
Fixed html of error messages.

brazabr committed:
Use Contents API to download single file from GitHub when possible.
